spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/dsl/Stream.java

```diff
@@ -0,0 +1,290 @@
+		return resource.getStatus();
+	}
+
+	private void createStreamDefinition() {
+		StringBuilder buffer = new StringBuilder();
+		this.definition = StringUtils.collectionToDelimitedString(applications, " | ");
+	}
+
+	public static class StreamBuilder {
+
+		private DataFlowOperations client;
+
+		private StreamBuilder(DataFlowOperations client) {
+			this.client = client;
+		}
+
+		public StreamNameBuilder name(String name) {
+			return new StreamNameBuilder(name, client);
+		}
+
+	}
+
+	public static class StreamNameBuilder {
+
+		private String name;
+
+		private List<StreamApplication> applications = new LinkedList<>();
+
+		private DataFlowOperations client;
+
+		private String definition;
+
+		private StreamNameBuilder(String name, DataFlowOperations client) {
+			this.client = client;
+			Assert.hasLength(name, "Stream name can't be empty");
+			this.name = name;
+		}
+
+		public SourceBuilder source(StreamApplication source) {
+			Assert.notNull(source, "Source application can't be null");
+			return new SourceBuilder(source.type(StreamApplication.StreamType.SOURCE),
+					this);
+		}
+
+		public SourceBuilder source(String name){
+			Assert.isTrue(StringUtils.hasLength(name), "Source application can't be null");
+			StreamApplication source = new StreamApplication(name);
+			return new SourceBuilder(source.type(StreamApplication.StreamType.SOURCE), this);
+		}
+
+		/**
+		 * Creates a Stream bypassing the fluent API and just using the provided
+		 * definition
+		 * @param definiton the Stream definition to use
+		 * @return A {@link Stream} object
+		 */
+		public StreamDefinitionBuilder definition(String definiton) {
+			Assert.hasLength(name, "Stream definition can't be empty");
+			this.definition = definiton;
+			return new StreamDefinitionBuilder(this.name, this.client, this.definition);
+		}
+
+		private Stream create() {
+			return new Stream(this.name, this.applications, this.definition, client)
+					.create();
+		}
+	}
+
+	public static class StreamDefinitionBuilder {
+
+		private String name;
+
+		private DataFlowOperations client;
+
+		private String definition;
+
+		private StreamDefinitionBuilder(String name, DataFlowOperations client,
+				String definition) {
+			this.name = name;
+			this.client = client;
+			this.definition = definition;
+		}
+
+		public Stream create() {
+			return new Stream(this.name, Collections.emptyList(), this.definition, client)
+					.create();
+		}
+	}
+
+	public static class SourceBuilder extends BaseBuilder {
+
+		private SourceBuilder(StreamApplication source, StreamNameBuilder parent) {
+			super(source, parent);
+		}
+
+		public ProcessorBuilder processor(StreamApplication processor) {
+			Assert.notNull(processor, "Processor application can't be null");
+			return new ProcessorBuilder(
+					processor.type(StreamApplication.StreamType.PROCESSOR), this.parent);
+		}
+
+		public ProcessorBuilder processor(String name) {
+			Assert.hasLength(name, "Stream definition can't be empty");
```

Comment:
Can we say, `Application name` instead of `Stream definition`?

Reply:
It looks like the same message is used in other places where the application name is used. If there is no specific reason for it, we can fix them as well.